Doc/library/itertools.rst

```diff
@@ -464,6 +465,25 @@ loops that truncate the stream.
                       pass
 
 
+.. function:: ilen(iterable)
+
+   Return the length of an *iterable* by counting iterations.
+
+   If *iterable* is an iterator, it is consumed.
```

Comment:
Maybe the mention to `len` is not needed then. If `islice` does not make the distinction, then we should not mention it.